arwen/elrondapi/managedConversions.go

```diff
@@ -100,20 +105,171 @@ func readESDTTransfer(
 	}, nil
 }
 
+// Converts a managed buffer of serialized data to a slice of ESDTTransfer.
+// The format is:
+// - token identifier handle - 4 bytes
+// - nonce - 8 bytes
+// - value handle - 4 bytes
+// Total: 16 bytes.
+func readESDTTransfers(
+	managedType arwen.ManagedTypesContext,
+	managedVecHandle int32,
+) ([]*vmcommon.ESDTTransfer, error) {
+	managedVecBytes, err := managedType.GetBytes(managedVecHandle)
+	if err != nil {
+		return nil, err
+	}
+	managedType.ConsumeGasForThisIntNumberOfBytes(len(managedVecBytes))
+
+	if len(managedVecBytes)%esdtTransferLen != 0 {
+		return nil, errors.New("invalid managed vector of ESDT transfers")
+	}
+
+	numTransfers := len(managedVecBytes) / esdtTransferLen
+	result := make([]*vmcommon.ESDTTransfer, 0, numTransfers)
+	for i := 0; i < len(managedVecBytes); i += handleLen {
+		esdtTransfer, err := readESDTTransfer(managedType, managedVecBytes[i:i+handleLen])
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, esdtTransfer)
+	}
+
+	return result, nil
+}
+
 // Serializes a vmcommon.ESDTTransfer object.
 func writeESDTTransfer(
-	host arwen.VMHost,
-	object *vmcommon.ESDTTransfer,
-) ([]byte, error) {
-	managedType := host.ManagedTypes()
-
-	tokenIdentifierHandle := managedType.NewManagedBufferFromBytes(object.ESDTTokenName)
-	valueHandle := managedType.NewBigInt(object.ESDTValue)
+	managedType arwen.ManagedTypesContext,
+	esdtTransfer *vmcommon.ESDTTransfer,
+	destinationBytes []byte,
+) {
+	tokenIdentifierHandle := managedType.NewManagedBufferFromBytes(esdtTransfer.ESDTTokenName)
+	valueHandle := managedType.NewBigInt(esdtTransfer.ESDTValue)
 
-	destinationBytes := make([]byte, 16)
 	binary.BigEndian.PutUint32(destinationBytes[0:4], uint32(tokenIdentifierHandle))
-	binary.BigEndian.PutUint64(destinationBytes[4:12], object.ESDTTokenNonce)
+	binary.BigEndian.PutUint64(destinationBytes[4:12], esdtTransfer.ESDTTokenNonce)
 	binary.BigEndian.PutUint32(destinationBytes[12:16], uint32(valueHandle))
+}
+
+// Serializes a list of ESDTTransfer one after the other into a byte slice.
+// The format is (for each ESDTTransfer):
+// - token identifier handle - 4 bytes
+// - nonce - 8 bytes
+// - value handle - 4 bytes
+// Total: 16 bytes.
+func writeESDTTransfersToBytes(
+	managedType arwen.ManagedTypesContext,
+	esdtTransfers []*vmcommon.ESDTTransfer,
+) []byte {
+	destinationBytes := make([]byte, esdtTransferLen*len(esdtTransfers))
+	dataIndex := 0
+	for _, esdtTransfer := range esdtTransfers {
+		writeESDTTransfer(managedType, esdtTransfer, destinationBytes[dataIndex:dataIndex+esdtTransferLen])
+		dataIndex += esdtTransferLen
+	}
+
+	return destinationBytes
+}
+
+type vmInputData struct {
+	destination []byte
+	function    string
+	value       *big.Int
+	arguments   [][]byte
+}
+
+func readDestinationValueFunctionArguments(
+	vmHost arwen.VMHost,
```

Comment:
i would prefer a rename to `host` everywhere in this file as this is how it's named all over arwen (please)

Reply:
done